src/cript/nodes/util/__init__.py

```diff
@@ -155,6 +155,23 @@ def __init__(self, uid: str):
 
 class _NodeDecoderHook:
     def __init__(self, uid_cache: Optional[Dict] = None):
+        """
+        Initialize the custom JSON object hook used for CRIPT node deserialization.
+
+        Parameters
+        ----------
+        uid_cache : Optional[Dict], optional
+            A dictionary to cache Python objects with shared UIDs, by default None.
+
+        Notes
+        -----
+        The `_NodeDecoderHook` class is used as an object hook for JSON deserialization,
+        handling the conversion of JSON nodes into Python objects based on their node types.
+        The `uid_cache` is an optional dictionary to store cached objects with shared UIDs
```

Comment:
The uid cache is not to avoid loops.

It is to never create two different python objects with the same uid.

So if you pass a JSON for a node with a UID that is present in the uid_cache, instead of creating a new python object for that node, the existing python object from the `uid_cache` is used and updated with the values from the JSON.

Reply:
updated it